src/store/modules/events/index.ts

```diff
@@ -494,27 +499,64 @@ const module: Module<EventsModuleState, RootState> = {
       const skip = (state.repetitions[key] || []).length;
 
       const response = await eventsApi.getLatestRepetitions(projectId, eventId, skip, limit);
-      const repetitions = response.data.project.event.repetitions;
-
-      /**
-       * Solution for not displaying both `userAgent` and `beautifiedUserAgent` addons
-       */
-      filterBeautifiedAddons(repetitions);
 
       if (originalEvent) {
         filterBeautifiedAddons([ originalEvent ]);
       }
 
-      repetitions.map(repetition => {
-        // save to the state
-        commit(MutationTypes.AddRepetitionPayload, {
-          projectId,
-          eventId,
-          repetition: {
+      /**
+       * If delta is present, apply delta to the event payload
+       */
+      const repetitions = response.data.project.event.repetitions.map(repetition => {
+        let processedRepetition: HawkEventRepetition;
+
+        const isNewDeltaFormat = !repetition.payload;
+
+        if (isNewDeltaFormat && originalEvent) {
+          const eventPayload = cloneDeep(originalEvent.payload);
+
+          /**
+           * If delta is present, apply delta to the event payload, otherwise set the event payload to the original event payload
+           */
+          processedRepetition = {
             ...repetition,
-            payload: originalEvent ? repetitionAssembler(originalEvent.payload, repetition.payload) as HawkEventPayload : repetition.payload,
-          },
-        });
+            payload: repetition.delta ? patch({
```

Comment:
same as below: it is expected to see `processRepetitionPayload` here as single source of truth

Reply:
done

Reply:
didn't notice